versionutils/version.go

```diff
@@ -20,77 +20,139 @@ var (
 	InvalidSemverVersionError = func(tag string) error {
 		return errors.Errorf("Tag %s is not a valid semver version, must be of the form vX.Y.Z[-rc#]", tag)
 	}
-	InvalidReleaseCandidateTag = func(tagAndBuildMetadata string) error {
-		return errors.Errorf("Semver tag %s is not valid release candidate (must be 'rc' followed by int, e.g. 'rc5')", tagAndBuildMetadata)
-	}
 )
 
 type Version struct {
-	Major            int
-	Minor            int
-	Patch            int
-	ReleaseCandidate int
-}
+	Major int
+	Minor int
+	Patch int
 
-func NewVersion(major, minor, patch int) *Version {
-	return NewRcVersion(major, minor, patch, 0)
+	// optional to support a version like "1.0.0-rc1", where "rc" is the label and "1" is the label version
+	// for comparisons:
+	//  - "1.0.0-rc1" is greater than "0.X.Y" and less than "1.0.0"
+	//  - "1.0.0-rc5" is greater than "1.0.0-rc1"
+	//  - "1.0.0-aX" is not greater than or less than "1.0.0-bY", except by convention
+	Label        string
+	LabelVersion int
 }
 
-func NewRcVersion(major, minor, patch, rc int) *Version {
+func NewVersion(major, minor, patch int, label string, labelVersion int) *Version {
 	return &Version{
-		Major:            major,
-		Minor:            minor,
-		Patch:            patch,
-		ReleaseCandidate: rc,
+		Major:        major,
+		Minor:        minor,
+		Patch:        patch,
+		Label:        label,
+		LabelVersion: labelVersion,
 	}
 }
 
 func (v *Version) String() string {
-	if v.ReleaseCandidate == 0 {
+	if v.LabelVersion == 0 {
```

Comment:
Should this be
```suggestion
	if v.LabelVersion == 0  || v.Label == 0 {
```
?

Reply:
no, if the label version is 0 then we print x.y.z instead of x.y.z-labela